infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/InfinispanRemoteDatastoreProvider.java

```diff
@@ -109,7 +115,9 @@
 	@Override
 	public void start() {
 		hotrodClient = HotRodClientBuilder.builder().withConfiguration( config, marshaller ).build();
-		hotrodClient.start();
+		// TODO temporary create another cache manager to handle remote script registration and invocation due to incompatibility of ProtoStreamMarshaller.
+		// When https://issues.jboss.org/browse/ISPN-8020 is closed, we could remove it and reuse the common hotrodClient.
+		scriptManager = HotRodClientBuilder.builder().withConfiguration( config, new GenericJBossMarshaller() ).build();
 		config = null; //no longer needed
```

Comment:
Are you stopping the scriptManager at some point?
Also, how come we don't need to start the hotrodClient anymore?

Reply:
I believe it start by default